### PR TITLE
[8] - feat(frontend): improves the UI and  adds export features

### PR DESCRIPTION
**Pull request by Kushdapush**
This PR does the following:

1.  Adds a cover image and UI formatting of the home page
2. Adds a feature to export the mean values in form of  CSV file
3. Adds a feature to export the charts to PNG

### CODE REVIEW

**Comment by sbhaya** [comment]
Screenshot 2024-08-31 at 9 32 10 PM

@Kushdapush Here is an enhancement request I would like to make. As the font-family used is "Noto Serif" on the current IWP website, is there a possibility that we can incorporate that font-family into our project?

**Comment by sbhaya** [comment]
Screenshot 2024-08-31 at 9 36 05 PM

@Kushdapush I would like to nudge you think of refining our "Return" label in the navigation bar that is more descriptive and clear, as to where the user is being led to once they click on it.

File: frontend/src/app/statistics/page.tsx [hunk 1]
```diff
@@ -71,10 +49,18 @@ export default function Statistics() {
...
+      !endingYear ||
+      !infoType
+    ) {
       alert("Please fill all the fields.");
       return;
     }
+    setGifState(true);
     console.log(state, district, parameter, startingYear, endingYear, infoType);
     axios
       .get("http://localhost:8080/api/statistics", {
```
Comment (sbhaya) [inline frontend/src/app/statistics/page.tsx:66]: @Kushdapush We will create separate environment files for production and development, define appropriate variables in each, and use these variables within our code accordingly. 
You may refer to this : [Reference link for setting up environment variables](https://github.com/vercel/next.js/discussions/12077#discussioncomment-92496)

File: frontend/src/components/MetDataSection/index.tsx [hunk 2]
```diff
@@ -3,50 +3,50 @@
...
+          <h1 className="mb-4 text-3xl font-extrabold leading-none tracking-normal text-white md:text-4xl lg:text-4xl">
             Met Data
           </h1>
-          <p className="mb-6 text-left font-normal text-gray-600 lg:text-xl   leading-7">
-          The India Water Portal serves as a treasure of information, housing meteorological datasets for all districts of India
-          spanning over the 20th Century `(1901-2002)`. This rich resource, encompassing 13 crucial climate parameters, offers 
-          invaluable insights for researchers, water resource managers, and anyone interested in understanding India &apos; s 
-          water story. This data can be useful in making rainwater harvesting and water balance estimates, in various research 
-          areas, climate change adaptation studies and more.
+          <p className="mb-6 text-center font-normal text-white lg:text-xl leading-7">
```
Comment (sbhaya) [inline frontend/src/components/MetDataSection/index.tsx:13]: Screenshot 2024-08-31 at 10 07 34 PM

@Kushdapush We can attempt to justify the text for improved readability and slightly reduce the font-weight of this description to enhance visual balance. Try 200 maybe.

Comment (Kushdapush) [reply frontend/src/components/MetDataSection/index.tsx:13]: @sbhaya i believe that making the text even lighter will make it more difficult to read, in the below Screenshot we can see that a white font weight 400 makes the text clearer as compared to white 300 against a blueish background

image

**Comment by Kushdapush** [comment]
> Screenshot 2024-08-31 at 9 36 05 PM
> @Kushdapush I would like to nudge you think of refining our "Return" label in the navigation bar that is more descriptive and clear, as to where the user is being led to once they click on it.

I am thinking of changing the text from "Return" to "Exit"

File: frontend/src/app/statistics/page.tsx (hunk 1 above)
Comment (Kushdapush) [reply frontend/src/app/statistics/page.tsx]: Alright, ill make respective env files for local, staging and production

**Comment by sbhaya** [comment]
> > Screenshot 2024-08-31 at 9 36 05 PM
> > @Kushdapush I would like to nudge you think of refining our "Return" label in the navigation bar that is more descriptive and clear, as to where the user is being led to once they click on it.
> 
> I am thinking of changing the text from "Return" to "Exit"

@Kushdapush That button click leads the user to "Background on the Meteorological Datasets", so naming it as Exit is also something not relevant. If it leads back to the landing page of India Water Portal, then it may become appropriate.

File: frontend/src/components/MetDataSection/index.tsx (hunk 2 above)
Comment (sbhaya) [reply frontend/src/components/MetDataSection/index.tsx:13]: @Kushdapush Agreeing with that remark of yours. The integration of **Noto Serif** font style and **justified text** will enhance the aesthetics of this text.

**Pull request merged**